manifests/resource/streamhost.pp

```diff
@@ -47,60 +47,25 @@
 #    ensure   => present,
 #  }
 define nginx::resource::streamhost (
-  $ensure                       = 'present',
-  $listen_ip                    = '*',
-  $listen_port                  = 80,
-  $listen_options               = undef,
-  $ipv6_enable                  = false,
-  $ipv6_listen_ip               = '::',
-  $ipv6_listen_port             = 80,
-  $ipv6_listen_options          = 'default ipv6only=on',
-  $proxy                        = undef,
-  $proxy_read_timeout           = $::nginx::proxy_read_timeout,
-  $proxy_connect_timeout        = $::nginx::proxy_connect_timeout,
-  $resolver                     = [],
-  $raw_prepend                  = undef,
-  $raw_append                   = undef,
-  $owner                        = $::nginx::global_owner,
-  $group                        = $::nginx::global_group,
-  $mode                         = $::nginx::global_mode,
+  Enum['absent', 'present'] $ensure       = 'present',
+  Variant[Array, String] $listen_ip       = '*',
+  Integer $listen_port                    = 80,
+  Optional[String] $listen_options        = undef,
+  Boolean $ipv6_enable                    = false,
+  Variant[Array, String] $ipv6_listen_ip  = '::',
+  Integer $ipv6_listen_port               = 80,
+  String $ipv6_listen_options             = 'default ipv6only=on',
+  $proxy                                  = undef,
```

Comment:
yes. But I want to add them in another PR. I want to keep the focus here on replacing validate functions. and in the next step adding types for all parameters + maybe improve existing ones.